tsfile/src/main/java/org/apache/iotdb/tsfile/read/common/block/TsBlock.java

```diff
@@ -357,52 +357,94 @@ public Object[] next() {
     }
   }
 
-  private class AlignedTsBlockIterator extends TsBlockSingleColumnIterator {
+  private class TsBlockAlignedRowIterator implements IPointReader, IBatchDataIterator {
 
-    private final int subIndex;
+    private int rowIndex;
 
-    private AlignedTsBlockIterator(int index, int subIndex) {
-      super(index);
-      this.subIndex = subIndex;
+    public TsBlockAlignedRowIterator(int rowIndex) {
+      this.rowIndex = rowIndex;
     }
 
     @Override
     public boolean hasNext() {
-      while (super.hasNext() && currentValue() == null) {
-        super.next();
-      }
-      return super.hasNext();
+      return rowIndex < positionCount;
     }
 
     @Override
     public boolean hasNext(long minBound, long maxBound) {
-      while (super.hasNext() && currentValue() == null) {
+      while (hasNext()) {
         if (currentTime() < minBound || currentTime() >= maxBound) {
           break;
         }
```

Comment:
wrong?

Reply:
Maybe, @liuminghui233 is trying to fix that, it's not my fault, the previous code was like that.